### DISCUSSION TITLE
DESeq2 normalized data or log2/vst normalized raw counts

### DISCUSSION THREAD

**Discussion by Aleksandra4496** [omnideconv/immunedeconv#174] [Q&A] [answered]
Hi,
I am performing RNAseq meta analysis and I would like to do immunedeconv methods (MCPcounter, xCell, Cibersort, ESTIMATE), but I don't know which data to import. With meta analysis I started with raw counts. My consideration is if I should to immunedeconv before batch effect correction for each dataset(if yes should I use raw counts or vst/log2 transformed counts)t, or after correction on metacohort (if yes, should I use DESeq2 norm data or raw counts or vst/log2 transformed data). 
Thank you in advance for your response and help!

**Comment by LorenzoMerotto** [comment, marked as answer]
Hello, 
With immunedeconv we recommend to use TPM-normalized data. Some of the methods (xCell, MPCcounter) work on the rank of the genes instead so normalization shouldn't have that big of an impact. I don't know how different you expect your cohorts to be, but to be safe I would use immunedeconv before the batch correction

**Reply by Aleksandra4496** [reply]
Dear Lorenzo, 
thank you for your response! Should I then use log2 transformation of the counts? 
All the best,
Aleksandra

**Reply by LorenzoMerotto** [reply]
Well, I wouldn't say they are intercheangable. If you cannot access the TPMs of your data I would suggest using the CPM
